checkov/cdk/checks/python/S3BucketKMSEncryption.yaml

```diff
@@ -1,25 +1,18 @@
 metadata:
-  version: 0.1
+  version: 0.2
   id: CKV_AWS_145
   name: Ensure that S3 buckets are encrypted with KMS by default
   category: ENCRYPTION
+  approach: define failing
 scope:
   languages:
     - python
 definition:
   - or:
-      - and:
-          - cond_type: pattern
-            operator: equals
-            value: "aws_cdk.aws_s3.Bucket(...)"
-          - cond_type: pattern
-            operator: not_equals
-            value: "aws_cdk.aws_s3.Bucket(..., encryption=aws_cdk.aws_s3.BucketEncryption.$ANY_VAR, ...)"
-      - and:
-          - cond_type: variable
-            variable: $ENCRYPTION
-            operator: regex_match
-            value: ^(S3_MANAGED|UNENCRYPTED)$
-          - cond_type: pattern
-            operator: equals
-            value: "aws_cdk.aws_s3.Bucket(..., encryption=aws_cdk.aws_s3.BucketEncryption.$ENCRYPTION, ...)"
+    - pattern: "aws_cdk.aws_s3.Bucket(<ANY>)"
```

Comment:
I think the first pattern is redundant, and only the second part is relevant. also adding a variable fore variable rendering could be helpful. (see example above)

Reply:
Without it, it doesn't catch a situation without encryption.

```python
      s3.Bucket(
            self,
            "example",
        )
```

I don't think this is able to be moved to metavariables until we have nested metavariables.